### PR TITLE
feat(across): send funds from ethereum to optimism

### PR DESCRIPTION
**Pull request by amateima**
https://app.shortcut.com/uma-project/story/3508/ability-to-send-funds-from-ethereum-to-optimism-in-across-frontend

This PR adds transfers from L1 to L2 using Optimism Bridge and follows the logic implemented by David [in this PR](https://github.com/across-protocol/across-frontend/pull/57), which has not been merged yet. I addressed your comments you left in that review. 

The L1 -> L2 Boba PR will follow the same structure, but I'm waiting for your feedback on this one before adding Boba

The only missing piece is calculating the gas fees, I'm going to make a separate PR for it

L2 deposit transactions made with Across:
- [USDC](https://optimistic.etherscan.io/tx/0x4704c9c84fbd5f3c92c08bd310ff04fc0c2b2fbd82e7e4bcb912a2d9e68478d0)
- [ETH](https://optimistic.etherscan.io/tx/0x99eda4acb9531d205020a27f8db893f37302c50e109790fb7aa5938df08d91ba)

### CODE REVIEW

File: src/components/CoinSelection/CoinSelection.tsx [hunk 1]
```diff
@@ -31,19 +25,32 @@ import {
...
+    return TOKENS_LIST[sendState.currentlySelectedFromChain.chainId];
+  }, [fromChain, toChain, sendState.currentlySelectedFromChain.chainId]);
   const { data: balances } = useBalances(
     {
       account: account!,
       chainId: sendState.currentlySelectedFromChain.chainId,
     },
     { skip: !account }
   );
+  const tokenBalanceMap = useMemo(() => {
```
Comment (amateima) [inline src/components/CoinSelection/CoinSelection.tsx:45]: `tokenList` supported by Optimism for deposits is not the same with `TOKENS_LIST`, this is why fetching balances by index doesn't work. I need to keep a map between tokens and balances. This is more of an improvisation and it has to be stored higher in the state

**Comment by daywiss** [comment]
Might need to hold off until the send-diff-chains branch is merged to master. Then get the branch epic/l1tl2 branch reset to master, rebase and pr against that branch

**Comment by amateima** [comment]
> Might need to hold off until the send-diff-chains branch is merged to master. Then get the branch epic/l1tl2 branch reset to master, rebase and pr against that branch

agree

**Review by Tulun — CHANGES_REQUESTED** [review]
Yeah I would say this is in a different epic. Re open this PR after we get the optimism stuff in from a different base.

**Comment by mrice32** [comment]
@amateima going to remove myself from review since we're gonna hold off on this PR. Feel free to re-add when you're ready for review!

**Pull request closed without merging**